src/Google.Cloud.Metadata.V1/DiskMetadata.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Google.Cloud.Metadata.V1
+{
+    /// <summary>
+    /// Metadata for a disk attached to a Google Compute Engine VM instance.
+    /// </summary>
+    /// <seealso cref="InstanceMetadata.Disks"/>
+    public sealed class DiskMetadata
+    {
+        /// <summary>
+        /// Gets the name of the disk, which could be null.
```

Comment:
No, this is a user-entered name when creating the disk. I've updated the comment.